numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
```

Comment:
It's done [here](https://github.com/numpy/numpy/blob/master/numpy/lib/tests/test_io.py#L318-L325).

My understanding is that if we were to use `mkstemp`, it would only guarantee that that file would be unique *until we removed it*. For this particular test, we need to get a unique filename (done here through `mkstemp`), then *delete* that file, so that we can start with a path with nothing on it. With a bare `mkstemp`, I think its conceivable (if very unlikely) that in between closing it and reopening it the OS might give it out to another program. However, if we ask for a unique *directory*, then we don't need to worry about anything in that directory being reassigned.

Maybe its unnecessary to use both `tempdir` and `mkstemp`, but I'm not sure its 100% safe to do this without the `tempdir` if we need not just an empty file, but a valid path without a file there.